addons/binding/org.openhab.binding.kostal.inverter/src/main/java/org/openhab/binding/kostal/inverter/WebscrapeHandler.java

```diff
@@ -0,0 +1,111 @@
+package org.openhab.binding.kostal.inverter;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.Iterator;
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.State;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class WebscrapeHandler extends BaseThingHandler {
+    private static Logger logger = LoggerFactory.getLogger(WebscrapeHandler.class);
+    private boolean active;
+    private SourceConfig config;
+    private List<ChannelConfig> channelConfigs;
+
+    public WebscrapeHandler(Thing thing) {
+        super(thing);
+        channelConfigs = new ArrayList<>();
+        channelConfigs.add(new ChannelConfig("acPower", "td", 4));
+        channelConfigs.add(new ChannelConfig("totalEnergy", "td", 7));
+        channelConfigs.add(new ChannelConfig("dayEnergy", "td", 10));
+        channelConfigs.add(new ChannelConfig("status", "td", 13));
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(SourceConfig.class);
+        updateStatus(ThingStatus.ONLINE);
+        ExecutorService executor = Executors.newSingleThreadExecutor();
+        executor.execute(new Runnable() {
+            @Override
+            public void run() {
+                while (active) {
+                    try {
+                        refresh();
+                    } catch (Exception e) {
+                        logger.warn("Error refreshing source " + getThing().getUID(), e);
```

Comment:
You should set the Thing status to OFFLINE with the exception message as a status message here - no need to do a warn logging then.